daemon/containerd/service.go

```diff
@@ -203,3 +225,14 @@ func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID st
 	// TODO(thaJeztah): include content-store size for the image (similar to "GET /images/json")
 	return usage.Size, usage.Size + snapShotSize, nil
 }
+
+// getContainerImageManifest safely dereferences ImageManifest.
+// ImageManifest can be nil for containers created with Docker Desktop with old
+// containerd image store integration enabled which didn't set this field.
+func getContainerImageManifest(ctr *container.Container) (ocispec.Descriptor, error) {
+	if ctr.ImageManifest == nil {
+		return ocispec.Descriptor{}, errdefs.InvalidParameter(errors.New("container is missing ImageManifest (probably created on old version), please recreate it"))
```

Comment:
We can probably create a tracking ticket to remove this after some time? (IIUC, this is a transitional situation for users that used older versions of the daemon with containerd-integration enabled)

Reply:
- opened https://github.com/moby/moby/issues/45408

Reply:
Yes, we want to remove this at some point.